app/views/layouts/application.html.erb

```diff
@@ -37,12 +37,21 @@
     <main role="main" class="container pb-5">
       <% flash.each do |type, message| %>
         <% next unless message.is_a? String %>
-        <div class="<% if !email_authentication_signed_in? %>col-md-6 mx-auto<% end %> alert <%= bootstrap_class_for_flash(type) %> alert-dismissable fade show">
-          <%= message %>
-          <button type="button" class="close" data-dismiss="alert" aria-label="Close">
-            <i class="fas fa-times" style="font-size: 85%; position: relative; bottom: 2px;"></i>
-          </button>
-        </div>
+          <% unless email_authentication_signed_in? %>
+            <div class="col-md-6 mx-auto">  
+          <% end %>
+          
+          <div class="mb-4 <% if !email_authentication_signed_in? %> <% end %> alert <%= bootstrap_class_for_flash(type) %> alert-dismissable fade show">
+            <%= message %>
+            <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+              <i class="fas fa-times" style="font-size: 85%; position: relative; bottom: 2px;"></i>
+            </button>
+          </div>
+          
+        <% unless email_authentication_signed_in? %>
+            </div>
+        <% end %>
```

Comment:
@dburka This section's got some changes with some awkward conditionals in it. Could you help me understand what we're trying to do here? I can then help think of nicer ways to code it up.

Reply:
This is to specifically format the warning messages on the login screen differently @harimohanraj89 . All other screens are full-width user interfaces and the login screen is a little window in the middle of the screen.

Reply:
Understood. If you're not signed in, wrap everything in a narrower div to line up with the little login windows 👍 

Sweet! There might be an elegant way to code that here, I'll push a commit and we can get this thing reviewed and shipped.

Reply:
Ok!